### PR TITLE
Add new cluster washer-control-cluster

### PR DESCRIPTION
**Pull request by crlonxp**

### CODE REVIEW

File: src/app/zap-templates/zcl/data-model/chip/washer-controls-cluster.xml [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+-->
+<configurator>
+  <domain name="CHIP"/>
+
+  <enum name="WasherControlsEnum" type="ENUM8">
```
Comment (bzbarsky-apple) [inline src/app/zap-templates/zcl/data-model/chip/washer-controls-cluster.xml:20]: This type does not exist in the spec.

File: src/app/zap-templates/zcl/data-model/chip/washer-controls-cluster.xml [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+    <cluster code="0x0061"/>
+    <item name="Spin" value="0x00"/>
+    <item name="Rinse" value="0x01"/>
+    <item name="Error"   value="0x03"/>
+  </enum>
+
+  <enum name="ErrorStateEnum" type="ENUM8">
```
Comment (bzbarsky-apple) [inline src/app/zap-templates/zcl/data-model/chip/washer-controls-cluster.xml:27]: What does this have to do with Washer Controls?

File: src/app/zap-templates/zcl/data-model/chip/washer-controls-cluster.xml [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+    <cluster code="0x0061"/>
+    <item name="NoError"                    value="0x00"/>
+    <item name="CommandInvalidInState"      value="0x01"/>
+  </enum>
+
+  <struct name="SpinSpeedIndex">
+    <cluster code="0x0061"/>
+    <item name="WasherControlsID"    type="WasherControlsEnum" optional="false"/>
+    <item name="WasherControlsLabel" type="CHAR_STRING" optional="false"/>
```
Comment (bzbarsky-apple) [inline src/app/zap-templates/zcl/data-model/chip/washer-controls-cluster.xml:36]: These field names do not match the spec.

File: src/app/zap-templates/zcl/data-model/chip/washer-controls-cluster.xml [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+    <cluster code="0x0061"/>
+    <item name="ErrorStateID" type="ErrorStateEnum" optional="false" />
+    <item name="ErrorStateLabel" type="CHAR_STRING" optional="false" length="64"/>
+    <item name="ErrorStateDetails" type="CHAR_STRING" optional="false" length="64"/>
+  </struct>
+
+  <cluster>
+    <domain>General</domain>
+    <name>Washer Controls</name>
+    <code>0x0061</code>
```
Comment (bzbarsky-apple) [inline src/app/zap-templates/zcl/data-model/chip/washer-controls-cluster.xml:49]: This cluster id does not match the spec.

**Review by bzbarsky-apple — CHANGES_REQUESTED** [review]
This XML seems mostly unrelated to the Washer Controls spec as it currently exists.  I checked for in-flight spec PRs that would affect that, and don't see any.

Please carefully go through the XML and make it match the spec, then request review.

File: src/app/zap-templates/zcl/data-model/chip/washer-controls-cluster.xml (hunk 3 above)
Comment (crlonxp) [reply src/app/zap-templates/zcl/data-model/chip/washer-controls-cluster.xml]: It's fixed. Please review again. Thank you.

File: src/app/zap-templates/zcl/data-model/chip/washer-controls-cluster.xml (hunk 2 above)
Comment (crlonxp) [reply src/app/zap-templates/zcl/data-model/chip/washer-controls-cluster.xml]: It's fixed. Please review again. Thank you.

File: src/app/zap-templates/zcl/data-model/chip/washer-controls-cluster.xml (hunk 1 above)
Comment (crlonxp) [reply src/app/zap-templates/zcl/data-model/chip/washer-controls-cluster.xml]: It's fixed. Please review again. Thank you.

File: src/app/clusters/washer-controls-server/washer-controls-server.cpp
[the remark this answers is not in the document]
Comment (crlonxp) [reply src/app/clusters/washer-controls-server/washer-controls-server.cpp:1]: I have updated it. Please review.  Thank you

File: src/app/zap-templates/zcl/data-model/chip/washer-controls-cluster.xml [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+    <name>Washer Controls</name>
+    <code>0x0053</code>
+    <define>WASHER_CONTROLS_CLUSTER</define>
+    <client init="false" tick="false">true</client>
+    <server init="false" tick="false">true</server>
+    <description>This cluster supports remotely monitoring and controling the different typs of functionality available to a washing device, such as a washing machine.</description>
+
+    <globalAttribute side="server" code="0xFFFD" value="1" />
+
+    <attribute side="server" code="0x0000" define="SPIN_SPEEDS"               type="ARRAY" entryType="SpinSpeedIndex"                               writable="false" isNullable="false" optional="false">SpinSpeeds</attribute>
```
Comment (bzbarsky-apple) [inline src/app/zap-templates/zcl/data-model/chip/washer-controls-cluster.xml:43]:
```suggestion
    <attribute side="server" code="0x0000" define="SPIN_SPEEDS"               type="ARRAY" entryType="SpinSpeedIndex"                               writable="false" isNullable="false" optional="true">SpinSpeeds</attribute>
```
It's feature-dependent, so optional.

File: src/app/zap-templates/zcl/data-model/chip/washer-controls-cluster.xml [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+    <attribute side="server" code="0x0001" define="SPIN_SPEED_CURRENT"        type="INT8U"                                min="0x00" max="0x1F"    writable="true" isNullable="true"  optional="true">SpinSpeedCurrent</attribute>
```
Comment (bzbarsky-apple) [inline src/app/zap-templates/zcl/data-model/chip/washer-controls-cluster.xml:44]:
```suggestion
    <attribute side="server" code="0x0001" define="SPIN_SPEED_CURRENT"        type="INT8U"                                min="0x00" max="0x0F"    writable="true" isNullable="true"  optional="true">SpinSpeedCurrent</attribute>
```
Spec says "0 to 15", no?

File: src/app/zap-templates/zcl/data-model/chip/washer-controls-cluster.xml [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+    <attribute side="server" code="0x0002" define="NUMBER_OF_RINSES"          type="INT8U"                                min="0x00" max="0x003" writable="true" isNullable="true"  optional="true">NumberOfRinses</attribute>
```
Comment (bzbarsky-apple) [inline src/app/zap-templates/zcl/data-model/chip/washer-controls-cluster.xml:45]: Where did this `max="0x003"` come from?

File: src/app/zap-templates/zcl/data-model/chip/washer-controls-cluster.xml [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+    <attribute side="server" code="0x0003" define="MAXRINSES"                 type="INT8U"                                                         writable="true"                    optional="false">MaxRinses</attribute>
```
Comment (bzbarsky-apple) [inline src/app/zap-templates/zcl/data-model/chip/washer-controls-cluster.xml:46]: This is not writable, but is optional, in the spec, no?

File: src/app/zap-templates/zcl/data-model/chip/washer-controls-cluster.xml [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+-->
+<configurator>
+  <domain name="CHIP"/>
+
+  <bitmap name="WasherControlsFeatures" type="BITMAP8">
```
Comment (bzbarsky-apple) [inline src/app/zap-templates/zcl/data-model/chip/washer-controls-cluster.xml:20]:
```suggestion
  <bitmap name="Feature" type="BITMAP8">
```

File: src/app/zap-templates/zcl/data-model/chip/washer-controls-cluster.xml [hunk 10]
```diff
@@ -0,0 +1,49 @@
...
+    <cluster code="0x0053"/>
+    <item name="Spin" value="0x00"/>
+    <item name="Rinse" value="0x01"/>
```
Comment (bzbarsky-apple) [inline src/app/zap-templates/zcl/data-model/chip/washer-controls-cluster.xml:23]: I don't think those will work right.  Please compare to other bitmaps....

File: src/app/zap-templates/zcl/data-model/chip/washer-controls-cluster.xml (hunk 4 above)
Comment (crlonxp) [reply src/app/zap-templates/zcl/data-model/chip/washer-controls-cluster.xml]: It's fixed. Please review. Thank you
